tests/test-team/template-mgmt-component-tests/email/template-mgmt-submit-email-page.component.ts

```diff
@@ -0,0 +1,148 @@
+  });
+
+  test('when user visits page, then page is loaded', async ({
+    page,
+    baseURL,
+  }) => {
+    const submitEmailTemplatePage = new TemplateMgmtSubmitEmailPage(page);
+
+    await submitEmailTemplatePage.loadPage(sessions.valid.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/submit-email-template/${sessions.valid.id}`
+    );
+
+    await expect(submitEmailTemplatePage.pageHeader).toHaveText(
+      `Submit '${sessions.valid.emailTemplateName}'`
+    );
+  });
+
+  test.describe('Page functionality', () => {
+    test('common page tests', async ({ page, baseURL }) => {
+      const props = {
+        page: new TemplateMgmtSubmitEmailPage(page),
+        id: sessions.valid.id,
+        baseURL,
+      };
+
+      await assertSkipToMainContent(props);
+      await assertNotifyBannerLink(props);
+      await assertLoginLink(props);
+      await assertGoBackLink({
+        ...props,
+        expectedUrl: `templates/preview-email-template/${sessions.valid.id}`,
+      });
+    });
+
+    test('when user submits form with, then the "Template submitted" page is displayed', async ({
+      page,
+    }) => {
+      const submitEmailTemplatePage = new TemplateMgmtSubmitEmailPage(page);
+
+      await submitEmailTemplatePage.loadPage(sessions.submit.id);
+
+      await submitEmailTemplatePage.clickSubmitTemplateButton();
+
+      await expect(page).toHaveURL(
+        new RegExp('/templates/email-template-submitted/email-(.*)')
+      );
+
+      const templateId = getAndStoreTemplateId(page.url());
+
+      const template = await templateStorageHelper.getTemplate(templateId!);
+
+      expect(template).toBeTruthy();
+    });
+  });
+
+  test.describe('Error handling', () => {
+    test('when user visits page with missing data, then an invalid session error is displayed', async ({
+      baseURL,
+      page,
+    }) => {
+      const submitEmailTemplatePage = new TemplateMgmtSubmitEmailPage(page);
+
+      await submitEmailTemplatePage.loadPage(sessions.empty.id);
+
+      await expect(page).toHaveURL(`${baseURL}/templates/invalid-session`);
+    });
+
+    test('when user submits form and returns, then an invalid session error is displayed', async ({
+      baseURL,
+      page,
+    }) => {
+      const submitEmailTemplatePage = new TemplateMgmtSubmitEmailPage(page);
+
+      await submitEmailTemplatePage.loadPage(sessions.submitAndReturn.id);
+
+      await submitEmailTemplatePage.clickSubmitTemplateButton();
+
+      getAndStoreTemplateId(page.url());
+
+      // Minor delay here to ensure session is deleted
+      // This test feels inherently flakey due to waiting on a session being deleted
+      // 3 seconds is more than ample time to have a session deleted.
+      await page.waitForTimeout(3000);
+
+      await submitEmailTemplatePage.loadPage(sessions.submitAndReturn.id);
+
+      await expect(page).toHaveURL(`${baseURL}/templates/invalid-session`);
```

Comment:
you could also increase time out on this assertation specifically as it auto retries